src/envoy/http/authn/http_filter_factory.cc

```diff
@@ -74,12 +79,38 @@ class AuthnFilterConfig : public NamedHttpFilterConfigFactory,
   std::string name() override { return kAuthnFactoryName; }
 
  private:
-  HttpFilterFactoryCb createFilter() {
-    ENVOY_LOG(debug, "Called AuthnFilterConfig : {}", __func__);
+  // Convert istio-authn::jwt to jwt_auth::jwt in protobuf format.
+  void convertJwtAuthFormat(
+      const ::istio::authentication::v1alpha1::Jwt& jwt_authn,
+      Http::JwtAuth::Config::AuthFilterConfig* proto_config) {
+    // Todo: when istio-authn::jwt diverges from jwt_auth::jwt,
+    // may need to convert more fields.
+    auto jwt = proto_config->add_jwts();
+    MessageUtil::jsonConvert(jwt_authn, *jwt);
+    jwt->set_jwks_uri_envoy_cluster(kJwtClusterName);
+  }
 
-    return [&](Http::FilterChainFactoryCallbacks& callbacks) -> void {
+  HttpFilterFactoryCb createFilter(FactoryContext& context) {
+    ENVOY_LOG(debug, "Called AuthnFilterConfig : {}", __func__);
+    std::shared_ptr<Http::IstioAuthn::JwtAuthnFactoryStore> jwt_factory_store =
+        std::make_shared<Http::IstioAuthn::JwtAuthnFactoryStore>(context);
+    // In POC, only inspect the first credential_rule
+    if (policy_.credential_rules_size() > 0 &&
```

Comment:
You'll need to iterate through all credential rules and methods, as well as peers' methods.

Reply:
Stay tuned. Will iterate through all credential rules and methods when rebasing the code.

Reply:
Done.